tests/components/laundrify/test_sensor.py

```diff
@@ -0,0 +1,104 @@
+"""Test the laundrify sensor platform."""
+
+from datetime import timedelta
+import json
+import logging
+from unittest.mock import AsyncMock, patch
+
+from laundrify_aio import LaundrifyDevice
+from laundrify_aio.exceptions import LaundrifyDeviceException
+import pytest
+
+from homeassistant.components.laundrify.const import DOMAIN, MANUFACTURER, MODELS
+from homeassistant.components.sensor import SensorDeviceClass
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_UNIT_OF_MEASUREMENT,
+    STATE_UNKNOWN,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.util import slugify
+from homeassistant.util.dt import utcnow
+
+from tests.common import MockConfigEntry, async_fire_time_changed, load_fixture
+
+
+@pytest.fixture(name="mock_device")
+def laundrify_sensor_fixture() -> LaundrifyDevice:
+    """Return a default Laundrify power sensor mock."""
+    # Load test data from machines.json
+    machine_data = json.loads(load_fixture("laundrify/machines.json"))[0]
+
+    mock_device = AsyncMock(spec=LaundrifyDevice)
+    mock_device.id = machine_data["id"]
+    mock_device.manufacturer = MANUFACTURER
+    mock_device.model = machine_data["model"]
+    mock_device.name = machine_data["name"]
+    mock_device.firmwareVersion = machine_data["firmwareVersion"]
+    return mock_device
+
+
+async def test_laundrify_sensor_init(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    mock_device: LaundrifyDevice,
+    laundrify_config_entry: MockConfigEntry,
+) -> None:
+    """Test Laundrify sensor default state."""
+    device_slug = slugify(mock_device.name, separator="_")
+
+    state = hass.states.get(f"sensor.{device_slug}_power")
+    assert state.attributes[ATTR_DEVICE_CLASS] == SensorDeviceClass.POWER
+    assert state.state == STATE_UNKNOWN
+
+    device = device_registry.async_get_device({(DOMAIN, mock_device.id)})
+    assert device is not None
+    assert device.name == mock_device.name
+    assert device.identifiers == {(DOMAIN, mock_device.id)}
+    assert device.manufacturer == mock_device.manufacturer
+    assert device.model == MODELS[mock_device.model]
+    assert device.sw_version == mock_device.firmwareVersion
+
+
+async def test_laundrify_sensor_update(
+    hass: HomeAssistant,
+    mock_device: LaundrifyDevice,
+    laundrify_config_entry: MockConfigEntry,
+) -> None:
+    """Test Laundrify sensor update."""
+    device_slug = slugify(mock_device.name, separator="_")
+
+    state = hass.states.get(f"sensor.{device_slug}_power")
+    assert state.state == STATE_UNKNOWN
+
+    with patch("laundrify_aio.LaundrifyDevice.get_power", return_value=95):
+        future = utcnow() + timedelta(minutes=2)
+        async_fire_time_changed(hass, future)
+        await hass.async_block_till_done()
```

Comment:
also use the freezer here

Reply:
Refactored in 0ef5e6f80539500bac04feb20aefc0a0fef01e19